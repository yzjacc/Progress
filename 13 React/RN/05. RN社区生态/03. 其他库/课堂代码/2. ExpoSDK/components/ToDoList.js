import { StyleSheet, View, Text } from "react-native";
import Input from "./Input";
import List from "./List";

export default function ToDoList() {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>待办事项</Text>
      <Input />
      <List />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    marginTop: 50,
  },
  title: {
    fontSize: 20,
    marginBottom: 10,
  },
});
