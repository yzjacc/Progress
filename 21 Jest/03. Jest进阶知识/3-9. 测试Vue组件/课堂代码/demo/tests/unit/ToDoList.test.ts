import { shallowMount } from "@vue/test-utils";
import ToDoList from "@/components/ToDoList.vue";

test("测试新增待办事项",async () => {
    const wrapper = shallowMount(ToDoList);

    // 生成快照
    expect(wrapper.element).toMatchSnapshot();

    const todo = wrapper.get('[data-test="todo"]');
    // 因为一开始只有一条待办事项
    expect(todo.text()).toBe('Learn Vue.js 3');

    // act
    // 接下来来测试新增
    await wrapper.get('[data-test="new-todo"]').setValue('New To Do Item');
    // 触发 submit 事件
    await wrapper.get('[data-test="form"]').trigger("submit");

    // assert
    expect(wrapper.findAll('[data-test="todo"]')).toHaveLength(2);
});
