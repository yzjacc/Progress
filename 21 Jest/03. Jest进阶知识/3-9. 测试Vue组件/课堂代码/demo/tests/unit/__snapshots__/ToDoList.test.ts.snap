// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试新增待办事项 1`] = `
<div>
  <!-- 渲染待办事项每一条项目 -->
  
  <div
    data-test="todo"
  >
    Learn Vue.js 3
  </div>
  
  <form
    data-test="form"
  >
    <input
      data-test="new-todo"
    />
  </form>
</div>
`;
